src/Services/RefundsService.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Liuv\Larapix\Services;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\ClientException;
+use Liuv\Larapix\Contracts\Features\RefundsContract;
+use Liuv\Larapix\ValueObjects\Refund;
+
+class RefundsService implements RefundsContract
+{
+
+    const BASE_API = 'https://api.openpix.com.br/api/openpix/v1';
+
+    private $client;
+
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
+    }
+
+
+    public function findById(string $refundId): array
+    {
+        $uri = sprintf(self::BASE_API . '/refund/%s', $refundId);
+
+        $response = $this->client->get($uri);
+
+        return json_decode($response->getBody(), true);
+    }
+
+    public function findAll(array $parameters = []): array
+    {
+        $uri = self::BASE_API . '/refund';
+
+        $response = $this->client->get($uri);
+
+        return json_decode($response->getBody(), true);
+    }
+
+    public function create(Refund $refund): array
+    {
+        
+        $uri = self::BASE_API . '/refund';
+        $response = $this->client->post($uri, [
+            'headers' => [
+                'Content-Type' => 'application/json',
+                'Accept' => 'application/json',
+            ],
+            'body' => json_encode($refund)
+        ]);
+        // TODO: decide how to differentiate any type of error
+
+        return json_decode($response->getBody(), true);
```

Comment:
This is being duplicated for every method. Can't we create a Trait or a sort of Abstract Class for it ?

Reply:
trait is better in this case, my opinion :D

Reply:
I don't really like traits on this scenario. Probably a Helper function inside and Abstract Class can be more useful.

Traits for simple stuff like that is not good to maintain.

Reply:
Yeah, I agree. Given the fact that we could use it for every Service class.